src/app/code/community/Zendesk/Zendesk/Model/Observer.php

```diff
@@ -97,6 +97,20 @@ public function saveConfig(Varien_Event_Observer $observer)
                 Mage::getModel('core/config')->saveConfig('contacts/email/recipient_email', $oldEmail, $scope, $scopeId);
             }
         }
+
+        // If the zendesk domain is not found in the web widget snippet (wrapped with quotes), generate it again
+        $zDomain = Mage::getStoreConfig('zendesk/general/domain', $storeCode);
+        $widgetSnippet = Mage::getStoreConfig('zendesk/frontend_features/web_widget_code_snippet', $storeCode);
+        // Case insensitive search with single and double quotes, still better performance than 1 regexp search
+        if(stripos($widgetSnippet, "'{$zDomain}'") === false && stripos($widgetSnippet, '"'.$zDomain.'"') === false) {
```

Comment:
These 2 conditions try to find the Zendesk Domain wrapped by single quotes or double quotes within the snippet string.
* I used `stripos` in case someone wrote their domain in caps.
* The `===` is required because (even if it'd be unlikely in this case, it's just a good habit) using `== false` would also match if the string starts with the string (the returned value would be 0), as `== false` is equivalent to `== 0`.

The reason I'm looking for a quoted string is to be sure that we are matching the full domain, preventing a case in which for example `domain.zendesk.com` would give a false negative if the previous domain was `mydomain.zendesk.com`, because `domain.zendesk.com` would be found in the snippet, but that doesn't mean it was the domain being used and that we don't need to regenerate the snippet.